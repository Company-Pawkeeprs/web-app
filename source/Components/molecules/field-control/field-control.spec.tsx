import { render, screen } from '@testing-library/react';
import { Formik } from 'formik';
import { describe, expect } from 'vitest';
import FieldControl from './field-control';

const Wrapper = ({ validate, ...props }: any) => (
    <Formik
        initialValues={{ [props.name]: '' }}
        onSubmit={() => {}}
        validate={validate}
    >
        <FieldControl {...props} />
    </Formik>
);

describe('FieldControl component (Unit)', () => {
    const fieldName = 'test-field';
    const fieldLabel = 'Test Field Label';
    const errorMessage = 'Test error message';

    it('FieldControl matches snapshot', () => {
        const { container } = render(
            <Wrapper name="test" label="Test Label" />,
        );
        expect(container).toMatchSnapshot();
    });

    it('should render without errors', () => {
        render(<Wrapper name={fieldName} />);
        expect(screen.getByTestId(`input-${fieldName}`)).toBeInTheDocument();
    });

    it('should render label correctly', () => {
        render(<Wrapper name={fieldName} label={fieldLabel} />);
        expect(screen.getByTestId(`label-${fieldName}`)).toHaveTextContent(
            fieldLabel,
        );
    });

    // it('should show error message with correct text when there is a validation error', async () => {
    //     const errorMessage = 'This field is required';
    //     const { rerender } = render(
    //         <Wrapper
    //             name={fieldName}
    //             validate={(value: string) => (value ? undefined : errorMessage)}
    //         />
    //     );
    //     const input = screen.getByTestId(`input-${fieldName}`);
    //     const errMessage = screen.getByTestId(`err-${fieldName}`);

    //     await userEvent.click(input);
    //     await userEvent.hover(input);

    //     expect(errMessage).toHaveTextContent(errorMessage);

    //     rerender(<Wrapper name={fieldName} disabledError={true} />);
    //     expect(errMessage).not.toBeVisible();
    // });
});
