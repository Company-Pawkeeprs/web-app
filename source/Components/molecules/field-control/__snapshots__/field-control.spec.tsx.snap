// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FieldControl component (Unit) > FieldControl matches snapshot 1`] = `
<div>
  <label
    class="mb-0 list-group-item fs-12 form-label"
    data-testid="label-test"
    for="test"
  >
    Test Label
  </label>
  <div
    class="position-relative mb-2 input-group"
  >
    <input
      class="form-control"
      data-testid="input-test"
      id="test"
      name="test"
      value=""
    />
  </div>
  <label
    class="font-bold list-group-item text-danger fs-11 text-center"
    data-testid="err-test"
    style="display: none;"
  />
</div>
`;
